nifi-server-api/src/main/java/org/apache/nifi/NiFiServer.java

```diff
@@ -33,6 +34,10 @@
 
     void setBundles(Bundle systemBundle, Set<Bundle> bundles);
 
+    void setNiFiProperties(NiFiProperties properties);
+
+    default void init() {}
```

Comment:
Rather than having two setters that must be called, and then having an init() method (which I'm guessing is expected to be called only after the setters?) perhaps it makes more sense to instead of an `initialize(NiFiProperties properties, Bundle systemBundle, Set<Bundle> additionalBundles);` method?

Reply:
Makes sense, the `setBundles` method was already there so I just kept the approach consistent, but since they're all called in the same place (including `setExtensionMapping`), I can just move everything to a single `initialize` method.